server/graphql/v1/mutations/collectives.js

```diff
@@ -291,7 +291,7 @@ export function editCollective(_, args, req) {
   }
 
   const newCollectiveData = {
-    ...omit(args.collective, ['location', 'type', 'ParentCollectiveId', 'data']),
```

Comment:
We should pick just the fields that we want, users should not be able to override the full `data` object